moveit_ros/planning/planning_components_tools/src/compare_collision_speed_checking_fcl_bullet.cpp

```diff
@@ -0,0 +1,354 @@
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Jens Petit */
+
+#include <moveit/planning_scene_monitor/planning_scene_monitor.h>
+#include <moveit/collision_plugin_loader/collision_plugin_loader.h>
+#include <moveit/collision_detection_bullet/collision_detector_allocator_bt.h>
+#include <moveit/collision_detection_fcl/collision_detector_allocator_fcl.h>
+#include <geometric_shapes/shape_operations.h>
+#include <random_numbers/random_numbers.h>
+
+#include <moveit/robot_model/robot_model.h>
+#include <moveit/utils/robot_model_test_utils.h>
+
+static const std::string ROBOT_DESCRIPTION = "robot_description";
+static const int MAX_SEARCH_FACTOR_CLUTTER = 3;
+static const int MAX_SEARCH_FACTOR_STATES = 30;
+
+enum class RobotStateSelector
+{
+  IN_COLLISION,
+  NOT_IN_COLLISION,
+  RANDOM,
+};
+
+enum class CollisionDetector
+{
+  FCL,
+  BULLET,
+};
+
+enum class CollisionObjectType
+{
+  MESH,
+  BOX,
+};
+
+/** \brief Clutters the world of the planning scene with random objects in a certain area around the origin. All added
+ *  objects are not in collision.
+*
+*   \param planning_scene The planning scene
+*   \param num_objects The number of objects to be cluttered
+*   \param CollisionObjectType Type of object to clutter (mesh or box) */
+void clutterWorld(planning_scene::PlanningScenePtr planning_scene, const size_t num_objects, CollisionObjectType type)
+{
+  ROS_INFO("Cluttering scene...");
+
+  random_numbers::RandomNumberGenerator num_generator = random_numbers::RandomNumberGenerator(123);
+
+  // allow all robot links to be in collision for world check
+  collision_detection::AllowedCollisionMatrix acm{ collision_detection::AllowedCollisionMatrix(
+      planning_scene->getRobotModel()->getLinkModelNames(), true) };
+
+  // set the robot state to home position
+  robot_state::RobotState& current_state{ planning_scene->getCurrentStateNonConst() };
+  collision_detection::CollisionRequest req;
+  current_state.setToDefaultValues(current_state.getJointModelGroup("panda_arm"), "home");
+  current_state.update();
+
+  // load panda link5 as world collision object
+  std::string name{ "" };
```

Comment:
no need to initialize a string

https://stackoverflow.com/questions/17738439/value-and-size-of-an-uninitialized-stdstring-variable-in-c

Reply:
changed in PR #1551